rspace/src/main/scala/coop/rchain/rspace/LMDBOps.scala

```diff
@@ -45,37 +45,8 @@ trait LMDBOps extends CloseOps {
       txn.commit()
       ret
     } catch {
-      case ex: Throwable =>
-        try {
-          txn.abort()
-        } catch {
-          //  /**
-          //   * Commits this transaction.
-          //   */
-          //  public void commit() {
-          //    checkReady();
-          //    (1) state = DONE;
-          //    checkRc( (2) LIB.mdb_txn_commit(ptr));
-          //  }
-          //   /**
-          //   * Aborts this transaction.
-          //   */
-          //  public void abort() {
-          //    (3) checkReady();
-          //    state = DONE;
-          //    LIB.mdb_txn_abort(ptr);
-          //  }
-          case ex: NotReadyException =>
-            ex.printStackTrace()
-            TxnOps.manuallyAbortTxn(txn)
-          // due to the way LMDBjava tries to handle txn commit
-          // IF the DB runs out of space AND this occurs while trying to commit a txn (2)
-          // the internal java state (STATE in Txn) will be set to DONE (1) before the Txn is committed
-          // (the exception happens in the attempt to commit)
-          // the abort action will interpret this as an invalid state (3)
-          // A fix is to ignore this exception as the original one is a valid cause
-          // and it still will cause txn.close in the finally block
-        }
+      case NonFatal(ex) =>
+        ex.printStackTrace
```

Comment:
Scala exception mechanism does not allow you to specify which exceptions are thrown by method. Thus you end up in situation like the one above where:
1. you re-throw exception for somebody else to deal with it,
2. but the type signature does not allow to specify that you are doing it (throwing exceptions)
3. thus the caller of your function might not handle the error, giving you bad times "it's rspace fault!"
4. so you end up printing out stack traces

A better approach is to return the error, inform the caller about possibilities of errors in type signature. That will be
1. return `Either[Throwable, R`] instead of `R`
2. or return F[R] (tagles final to the rescue) with `ApplicativeError` instance

Regardless wich approach you chose, the type signature will change, thus the usage of the `withTxn` will change - users of this method will have to handle the possible error. This would be a change that would cut across the whole project, but its step in the right direction.

Other then that changes below LGTM, but I would consider my comment above with future changes